### PR TITLE
Dread patcher updates

### PR DESCRIPTION
**Pull request by duncathan**
- boss pickups
 - progressive items
 - PBs require mains

addresses some of the requirements to finish #2695

### CODE REVIEW

**Comment by duncathan** [comment]
checks will probably fail because open-dread-rando 0.4.0 doesn't exist yet

File: randovania/games/dread/patcher/open_dread_patcher.py [hunk 1]
```diff
@@ -131,29 +145,77 @@ def _pickup_detail_for_target(detail: ExportedPickupDetails) -> Optional[dict]:
...
+            if resources[0]["item_id"] == "ITEM_WEAPON_POWER_BOMB_MAX":
+                resources = [resources[-1]]
+
+            pickup_node = db.world_list.node_from_pickup_index(detail.index)
+            pickup_type = pickup_node.extra.get("pickup_type", "actor")
+
+            hud_text = detail.hud_text[0]
+            if len(detail.hud_text) > 1:
+                mapping = {
+                    "Varia Suit acquired.": "Progressive Suit acquired.",
```
Comment (henriquegemignani) [inline randovania/games/dread/patcher/open_dread_patcher.py:185]: ... there's _definitely_ a better way to do this 🤔

File: randovania/games/dread/patcher/open_dread_patcher.py [hunk 2]
```diff
@@ -197,3 +259,7 @@ def patch_game(self, input_file: Optional[Path], output_file: Path, patch_data:
         with output_file.joinpath("patcher.json").open("w") as f:
             json.dump(patch_data, f, indent=4)
         open_dread_rando.patch(input_file, output_file, patch_data)
+
+class DreadAcquiredMemo(dict):
```
Comment (henriquegemignani) [inline randovania/games/dread/patcher/open_dread_patcher.py:263]: Was this so we could change the text later on?

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

File: randovania/games/dread/patcher/open_dread_patcher.py (hunk 2 above)
Comment (duncathan) [reply randovania/games/dread/patcher/open_dread_patcher.py:263]: it differs from the prime message in exactly one way: dread acquisitions end with a period, not an exclamation mark

File: randovania/games/dread/layout/preset_describer.py [hunk 3]
```diff
@@ -2,8 +2,8 @@
 
 from randovania.games.dread.layout.dread_configuration import DreadConfiguration
 from randovania.layout.base.major_items_configuration import MajorItemsConfiguration
-from randovania.layout.preset_describer import format_params_base, fill_template_strings_from_tree, has_shuffled_item, \
-    has_vanilla_item
+from randovania.layout.preset_describer import _format_params_base, fill_template_strings_from_tree, has_shuffled_item, \
```
[the remark this answers is not in the document]
Comment (duncathan) [reply randovania/games/dread/layout/preset_describer.py:5]: I was wondering why that was a conflict!

File: randovania/games/dread/patcher/open_dread_patcher.py (hunk 1 above)
Comment (duncathan) [reply randovania/games/dread/patcher/open_dread_patcher.py:185]: I couldn't find how echoes does it so I just did the first thing I knew would work

File: randovania/games/dread/patcher/open_dread_patcher.py (hunk 2 above)
Comment (henriquegemignani) [reply randovania/games/dread/patcher/open_dread_patcher.py:263]: Oh. Though I'm planning on changing these to match the vanilla style of the popups anyway

File: randovania/games/dread/patcher/open_dread_patcher.py (hunk 1 above)
Comment (henriquegemignani) [reply randovania/games/dread/patcher/open_dread_patcher.py:185]: Echoes has a different text depending on which one you get.

My thought would be to add the original `PickupEntry` to `ExportedPickupDetails`, and then `if len(detail.hud_text) > 1` you do `f"{pickup.name} acquired."`.

File: randovania/games/dread/patcher/open_dread_patcher.py (hunk 2 above)
Comment (duncathan) [reply randovania/games/dread/patcher/open_dread_patcher.py:263]: this implementation matches the vanilla style in dread

Comment (henriquegemignani) [reply randovania/games/dread/patcher/open_dread_patcher.py:263]: Not for expansions :)

Comment (duncathan) [reply randovania/games/dread/patcher/open_dread_patcher.py:263]: ah, fair. can just add those to the dict ofc

**Review by henriquegemignani — APPROVED** [review]

**Pull request merged**